src/platform/CYW30739/DiagnosticDataProviderImpl.h

```diff
@@ -39,11 +39,13 @@ class DiagnosticDataProviderImpl : public DiagnosticDataProvider
 
     // ===== Methods that implement the PlatformManager abstract interface.
 
+    bool SupportsWatermarks() override { return true; }
```

Comment:
This should be returning false, given that this feature is not actually implemented.